tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -84,19 +85,21 @@ def validate(self):
         return self
 
     def process(self, resources):
-        session = utils.local_session(self.manager.session_factory)
-        client = self.manager.get_client('azure.mgmt.resource.ResourceManagementClient')
+        self.session = utils.local_session(self.manager.session_factory)
+        self.client = self.manager.get_client('azure.mgmt.resource.ResourceManagementClient')
+        with self.executor_factory(max_workers=3) as w:
```

Comment:
Let's move the constants out to variables

Reply:
The session and client are used in a shared function not in a class function.